src/views/List.jsx

```diff
@@ -1,13 +1,50 @@
+import { useEffect, useState } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data, listToken }) {
+	const [searchTerm, setSearchTerm] = useState('');
+	const [filteredData, setFilteredData] = useState([]);
+
+	// Function to update the filteredData based on the searchTerm
+	const updateFilteredData = () => {
+		const filteredItems = data.filter((item) => {
+			if (item.name) {
+				return item.name.toLowerCase().includes(searchTerm.toLowerCase());
```

Comment:
Nice thinking running `toLowerCase` on the `item.name` and the `searchTerm`! Another very helpful method we could tack onto these would be `trim()`, which would allow additional searches to get through to their intended targets. I've included a before and after example below of it in practice.

### without trim()
![without trim()](https://github.com/the-collab-lab/tcl-65-smart-shopping-list/assets/38340128/6dd9ff79-67d8-440f-99cf-156f8ebc9290)

### with trim()
![with trim()](https://github.com/the-collab-lab/tcl-65-smart-shopping-list/assets/38340128/4c52b902-5875-4cbf-a30b-073c5b5797fe)


_Bonus: If we really wanted, we could create a reusable `text` or `format` util method that accepts a string and returns a newly formatted string with these methods applied, but that one's entirely up to you._ :-)